mmv1/products/bigquery/terraform.yaml

```
@@ -109,6 +118,15 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         vars:
           private: "private"
           public: "public"
+      - !ruby/object:Provider::Terraform::Examples
+        name: "bigquery_dataset_access_authorized_routine"
+        skip_test: true  # not importable
+        primary_resource_type: "google_bigquery_dataset_access"
+        primary_resource_id: "authorized_routine"
+        vars:
+          private_dataset: "private-dataset"
```

Tests are currently failing with the error:

```
Invalid dataset ID "tf-test-public-dataset7wabmdpj1o". Dataset IDs must be alphanumeric (plus underscores) and must be at most 1024 characters long
```

Since it turns out dataset only allows `_`, not `-`, all dataset id vars need to be converted to use `_`.

```suggestion
          private_dataset: "private_dataset"
```

---

@melinath 
Thank you for reviewing.
I should have noticed that "-" could not be used as dataset names.
Sorry for bothering you.

Finally, the failed test has passed!
I appreciate your advices and supports.